hydra_client/connection/json_connection.py

```diff
@@ -90,12 +116,24 @@ def args_to_json_object(self, *args):
                 yield arg
             elif isinstance(arg, (int, float)):
                 yield arg
+            elif isinstance(arg, datetime):
+                #yield arg
+                yield datetime.strftime(arg, self.dateformat)
             elif isinstance(arg, hb.JSONObject):
                 yield arg
             elif isinstance(arg, collections.Mapping):
                 yield hb.JSONObject(arg)
             elif isinstance(arg, collections.Iterable):
-                if len(arg) > 0 and isinstance(arg[0], (six.string_types, int, float)):
+                arg = list(arg) ## in case it's a set or something that doesnt support indexing
+                if len(arg) > 0 and isinstance(arg[0], (six.string_types, int, float, datetime)):
```

Comment:
Is the question here really "Is `arg[0]` serializable?" rather than "Is it an instance of the following...?"

If so, could make a small util func which tests this predicate directly, e.g...

```
def isSerializable(arg):
    try:
        json.dumps(arg)
    except:
        return False
    return True
```
...and then test with...
```
if len(arg) > 0 and isSerializable(arg[0]):
```

Reply:
No, it's not necessarily about being serilaizeable. Datetimes are not serialieable, so they need to be converted to strings. The isSerializable criteria would fail in this case, which we don't want.